api/tasks.py

```diff
@@ -4,17 +4,19 @@
 from wordpress.user import WPUser
 from lazona_connector.celery import app
 
-
 @app.task
 def create_delivery(order):
-    delivery = Client(order).create_delivery()
-
-    if type(delivery).__name__ == 'Error':
-        return delivery.to_dict()
-    else:
-        delivery.print_pdf()
-        return {'success': 'all good'}
-
+    deliveries_by_vendor = Client(order).create_delivery()
+    deliveries_success = []
+    deliveries_error = []
+    for delivery in deliveries_by_vendor:
+        if type(delivery).__name__ == 'Error':
+            # TODO: logger info about the error
+            deliveries_error.append(delivery)
+        else:
+            deliveries_success.append(delivery)
+            delivery.send_mail_to_vendor()
+    return {'success': [deliveries_success], 'error': [deliveries_error]}
```

Comment:
I'm not sure this will provide any value now. Perhaps each list count but other than that, it feels YAGNI to me. It's something we don't have the need for.

Reply:
Yeah, not for the moment, it can be deleted. I was thinking for the future when we will need to update the status of the WC order, to alert those vendors and customer whose deliveries failed.